tracer/src/Datadog.Trace/Ci/CiEnvironment/GitCommandGitInfoProvider.cs

```diff
@@ -0,0 +1,144 @@
+    }
+
+    protected override bool TryGetFrom(DirectoryInfo gitDirectory, out IGitInfo? gitInfo)
+    {
+        var localGitInfo = new GitInfo
+        {
+            SourceRoot = gitDirectory.Parent?.FullName
+        };
+
+        gitInfo = localGitInfo;
+
+        try
+        {
+            // Ensure we have permissions to read the git directory
+            var safeDirectory = ProcessHelpers.RunCommand(
+                new ProcessHelpers.Command(
+                    cmd: "git",
+                    arguments: $"config --global --add safe.directory {gitDirectory.FullName}",
+                    workingDirectory: gitDirectory.FullName,
+                    useWhereIsIfFileNotFound: true));
+            if (safeDirectory?.ExitCode != 0)
+            {
+                localGitInfo.Errors.Add($"Error setting safe.directory: {safeDirectory?.Error}");
+            }
+
+            // Get the repository URL
+            var repositoryOutput = ProcessHelpers.RunCommand(
+                new ProcessHelpers.Command(
+                    cmd: "git",
+                    arguments: "ls-remote --get-url",
+                    workingDirectory: gitDirectory.FullName,
+                    useWhereIsIfFileNotFound: true));
+            if (repositoryOutput?.ExitCode == 0)
+            {
+                localGitInfo.Repository = repositoryOutput.Output.Trim();
+            }
+            else
+            {
+                localGitInfo.Errors.Add($"Error getting repository URL: {repositoryOutput?.Error}");
+            }
+
+            // Get the branch name
+            var branchOutput = ProcessHelpers.RunCommand(
+                new ProcessHelpers.Command(
+                    cmd: "git",
+                    arguments: "rev-parse --abbrev-ref HEAD",
+                    workingDirectory: gitDirectory.FullName,
+                    useWhereIsIfFileNotFound: true));
+            if (branchOutput?.ExitCode == 0 && branchOutput.Output.Trim() is { Length: > 0 } branchName && branchName != "HEAD")
+            {
+                localGitInfo.Branch = branchName;
+            }
+            else
+            {
+                localGitInfo.Errors.Add($"Error getting branch name: {branchOutput?.Error}");
+            }
+
+            // Get the remaining data from the log -1
+            var gitLogOutput = ProcessHelpers.RunCommand(
+                new ProcessHelpers.Command(
+                    cmd: "git",
+                    arguments: """log -1 --pretty='%H|,|%at|,|%an|,|%ae|,|%ct|,|%cn|,|%ce|,|%B'""",
+                    workingDirectory: gitDirectory.FullName,
+                    useWhereIsIfFileNotFound: true));
+            if (gitLogOutput?.ExitCode != 0)
+            {
+                localGitInfo.Errors.Add($"Error getting git log: {gitLogOutput?.Error}");
+                return false;
+            }
+
+            var gitLogDataArray = gitLogOutput.Output.Trim().Split(["|,|"], StringSplitOptions.None);
```

Comment:
You can probably parse this more efficiently (avoiding the allocations) with `StringExtensions.SplitIntoSpans()`, but not _required_ for this PR

Reply:
we need to change that extension to accept string and not only a `char` as a separator.

Reply:
Oh yeah 🙁 meh then 😄